### PR TITLE
refactor: switch Awilix container injection mode to proxy

### LINKED ISSUE #497: Change Awilix container injection mode from `CLASSIC` to `PROXY`
### Context

- #360

The `CLASSIC` injection mode from Awilix comes with a great DX - adding the dependency in a constructor based on the registered name will automatically load it into the desired service:

https://github.com/interledger/web-monetization-extension/blob/8d47b3ec6642f94223e10574e821ee0d3982bdef/src/background/services/background.ts#L30-L41

But this comes with it's own cons - during the build step we are not able to mangle during minification, because Awilix will try to load the registered name (i.e. `events`, `monetizationService`. As per Awilix documentation:

> Don't use CLASSIC if you minify your code! We recommend using CLASSIC in Node and PROXY in environments where minification is needed.

### Todos

- [ ] Change Awilix container injection mode to PROXY in the background
- [ ] Change Awilix container injection mode to PROXY in the content script

Note:

The constructors will need to be updated:

```ts
class Service {
  private dependency;
  
  constructor({ dependency })  {
    Object.assign(this, { dependency });
  }
}
```

or

```ts
class Service {
  private dependency;
  
  constructor({ dependency })  {
    this.dependency = dependency;
  }
}
```

### PR DESCRIPTION
**Pull request by raducristianpopa**
## Context

Closes #497.

## Changes proposed in this pull request

Switching to `PROXY`, will allow us to do mangling during minification.

### CODE REVIEW

**Comment by raducristianpopa** [comment]
<h2>Extension builds preview</h2>

<table role="table">
<thead>
  <tr>
    <th>Name</th>
    <th>Link</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td align="center">Latest commit</td>
    <td align="center">23ac24ab2dc5c32679919cac26e9d363233ee768</td>
  </tr>
  <tr>
    <td align="center">Latest job logs</td>
    <td align="center"><a href="https://github.com/interledger/web-monetization-extension/actions/runs/10346127340">Run #10346127340</a></td>
  </tr>
  <tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/27057400428/artifacts/1800279723">Download</a></td></tr><tr><td align="center">Badge</td><td align="center"><a href="https://github.com/interledger/web-monetization-extension/suites/27057400428/artifacts/1800279726">Download</a></td></tr>
</tbody>
</table>

File: src/background/services/background.ts [hunk 1]
```diff
@@ -23,22 +23,45 @@ import {
...
-    private openPaymentsService: OpenPaymentsService,
-    private monetizationService: MonetizationService,
-    private storage: StorageService,
-    private logger: Logger,
-    private tabEvents: TabEvents,
-    private sendToPopup: SendToPopup,
-    private events: EventsService,
-    private heartbeat: Heartbeat
-  ) {}
+  private browser: Browser
```
Comment (sidvishnoi) [inline src/background/services/background.ts:32]: How about this?
```suggestion
  private browser: Cradle['browser']
```

File: src/background/services/deduplicator.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { Logger } from '@/shared/logger'
+import type { Cradle } from '../container'
```
Comment (sidvishnoi) [inline src/background/services/deduplicator.ts:1]:
```suggestion
import type { Cradle } from '@/background/container'
```

**Review by sidvishnoi — APPROVED** [review]
Mangle everything!

**Pull request merged**